1-Vanilla/main.js

```diff
@@ -0,0 +1,231 @@
+      text: "Go to Wix offices",
+      status: "unfinished",
+      isEditMode: false,
+      id: 9,
+    },
+    {
+      text: "Watch Deni Avdjia game",
+      status: "unfinished",
+      isEditMode: false,
+      id: 10,
+    },
+    {
+      text: "Go to step 2",
+      status: "unfinished",
+      isEditMode: false,
+      id: 11,
+    },
+    {
+      text: "Infra webinar at 3",
+      status: "unfinished",
+      isEditMode: false,
+      id: 12,
+    },
+    {
+      text: "Pay the bills",
+      status: "unfinished",
+      isEditMode: false,
+      id: 13,
+    },
+    {
+      text: "Contribute code to open source",
+      status: "unfinished",
+      isEditMode: false,
+      id: 14,
+    },
+    {
+      text: "Finish Velo site",
+      status: "unfinished",
+      isEditMode: false,
+      id: 15,
+    },
+  ];
+  
+  const loadToDoList = (list) => {
+    const ul = document.getElementById("todoList");
+    ul.innerHTML = "";
+    list.forEach((item) => {
+      ul.appendChild(createItem(list, item));
+    });
+  };
+  
+  const addTodo = () => {
+    const inputText = document.getElementById("addTodo").value;
+    list.push({
+      text: inputText,
+      status: "unfinished",
+      id: list.length + 1,
+    });
+    loadToDoList(list);
+  };
+  
+  const removeTodo = (list, item) => {
+    list = list.filter(function (ele) {
+      return ele.id != item.id;
+    });
+  
+    loadToDoList(list);
+  };
+  
+  const editTodo = (list, item) => {
+    list = list.map((ele) => {
+      if (ele.id === item.id) {
+        ele.isEditMode = !ele.isEditMode;
+      }
+      return ele;
+    });
+    loadToDoList(list);
+  };
+  
+  const toggleTodo = (item) => {
+    const li = document.getElementById(item.id);
+    const text = li.children[0].children[1];
+    if (item.status === "finished") {
+      text.style["text-decoration"] = "none";
+      item.status = "unfinished";
+    } else {
+      text.style["text-decoration"] = "line-through";
+      item.status = "finished";
+    }
+  };
+  
+  const createButton = (cb, icon, className) => {
+    const button = document.createElement("button");
+    button.innerHTML = `<i class="${icon}" aria-hidden="true"></i>`;
+    button.classList.add(className);
+    button.onclick = cb;
+    button.style.cursor = "pointer";
+    return button;
+  };
+  
+  const createItem = (list, item) => {
+    const li = document.createElement("li");
+    li.id = item.id;
+  
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkbox.onclick = () => toggleTodo(item);
+  
+    const spanText = document.createElement("span");
+    spanText.classList.add("spanText");
+    spanText.appendChild(checkbox);
+  
+    if (item.isEditMode) {
+      //edit model
+      const inputText = document.createElement("input");
+      inputText.type = "text";
+      inputText.value = item.text;
+      inputText.classList.add("edit-todo");
+      inputText.id = `${item.id}-editTodo`;
```

Comment:
[nice-to-have] I suggest having the generating and parsing of `id`s in centralized utility functions so that it'll be more flexible for changes in the future, and reusable of course

Reply:
Alright, currently I use it only in 1 place, so I think that when I'll use it in other places in the code I'll will build a function for that.